test/framework/src/main/java/org/elasticsearch/cluster/coordination/AbstractCoordinatorTestCase.java

```diff
@@ -219,11 +219,12 @@ protected static int defaultInt(Setting<Integer> setting) {
     // 3. master sends out PublishRequests to nodes
     // 4. master receives PublishResponses from nodes
     // 5. master sends ApplyCommitRequests to nodes
-    // 6. nodes apply committed cluster state
-    // 7. master receives ApplyCommitResponses
-    // 8. apply committed state on master (last one to apply cluster state)
-    // 9. complete the publication listener back on the master service thread
-    public static final int CLUSTER_STATE_UPDATE_NUMBER_OF_DELAYS = 9;
+    // 6. nodes deserialize committed cluster state
+    // 7. nodes apply committed cluster state
+    // 8. master receives ApplyCommitResponses
+    // 9. apply committed state on master (last one to apply cluster state)
+    // 10. complete the publication listener back on the master service thread
+    public static final int CLUSTER_STATE_UPDATE_NUMBER_OF_DELAYS = 10;
```

Comment:
:+1: except that this extra step is happening between steps 3 & 4 in the old list. `org.elasticsearch.cluster.coordination.ApplyCommitRequest` only carries the term and version, it's `org.elasticsearch.cluster.coordination.PublishRequest` which carries the state that's being published.